compiler/src/utils/config.re

```diff
@@ -403,6 +403,14 @@ let optimization_level =
     Level_three,
   );
 
+let memory_base =
+  opt(
+    ~doc="Set the start address for the Grain runtime heap.",
+    ~names=["memory-base"],
+    ~conv=Cmdliner.Arg.int,
+    0x400,
+  );
```

Comment:
Food for thought: I'm not sure if we ever expose what the default value is, but I wonder if someone seeing this will think that the low 1k is free game (and it isn't). That of course leads to the next important thing: in `compcore.re` and `link.re`, there's this Binaryen setting that'll need to be toggled off if this flag is ever set, `Settings.set_low_memory_unused(false)` to tell Binaryen to not write data to that region. That makes me wonder if this should instead be `option(int)`, with the default as `None`. If we did that, then we can choose to start the heap at 0x400 and let Binaryen own that low 1k, and when the user sets a value, we know to turn off that setting. That'd also distinguish the `--memory-base=1024` case from the default. I'm not married to that approach, but I think it could make sense here.

Reply:
Instead of being an `option(int)`, why not just have it be `if memory_base < 0x400`?

Reply:
If you supply the flag at all, we'll need to do that binaryen change because you're expecting to be able to use that memory yourself.

Reply:
Ah, I understand now. If you think it's important, we can do that, but I really think that we should just document it appropriately so that users of this flag (which probably represent a very small group) know not to touch `0x0`-`0x400`. Personally, I think that would be better than adding potentially strange side effects to the flag (like, as a user, I'd be sort of surprised if `--memory-base 0x1200` caused that binaryen flag to get switched)

Reply:
Yeah, it's pretty important. Often users don't really have any control over that memory with them needing to reserve that space for the host to use it, like in Wasm-4's use case.